packages/peregrine/lib/talons/AddToCartDialog/addToCartDialog.gql.js

```diff
@@ -23,9 +22,11 @@ const GET_PRODUCT_DETAIL = gql`
                         }
                     }
                 }
+                # eslint-disable-next-line @graphql-eslint/require-id-when-available
                 ... on ConfigurableProduct {
+                    # eslint-disable-next-line @graphql-eslint/require-id-when-available
                     configurable_options {
-                        id
+                        uid
```

Comment:
By default, the Apollo client uses the `__typename` and `id` field for caching so we will also need to update the fields that Apollo client uses to cache ConfigurableProductOptions objects from `id` to `uid`. 

See: [Apollo policies file](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/Apollo/policies/index.js)

We should also update the mock data in the tests to reflect the new format